### PR TITLE
Temporary fixes for build errors

### PR DESCRIPTION
**Pull request by vamsi-amazon**
### Description
* Bug Issue for week, weekoftheyear fix : https://github.com/opensearch-project/sql/issues/2477
* Enhancement request issue for RFC3339_LENIENT dateformat issue: https://github.com/opensearch-project/sql/issues/2478
 * Added temporary fixes for above two issues for unblocking other PRs as it is taking longer time to fix them.
 * Included latest grammar files.
 * Excluded ANTLR generated files from spotless check.
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by noCharger** [comment]
Thanks Vamsi, just wanna link this fix to the issue for tracking https://github.com/opensearch-project/sql/issues/2468

**Review by YANG-DB — APPROVED** [review]

File: spark/src/main/antlr/FlintSparkSqlExtensions.g4 [hunk 1]
```diff
@@ -26,6 +26,7 @@ skippingIndexStatement
     | refreshSkippingIndexStatement
     | describeSkippingIndexStatement
     | dropSkippingIndexStatement
+    | vacuumSkippingIndexStatement
```
Comment (noCharger) [inline spark/src/main/antlr/FlintSparkSqlExtensions.g4:29]: Why do we need to update .g4?

**Review by noCharger — APPROVED** [review]
Overall looks good. Thanks Vamsi

File: core/src/test/java/org/opensearch/sql/expression/datetime/DateTimeFunctionTest.java [hunk 2]
```diff
@@ -1236,17 +1236,20 @@ public void testWeekOfYearWithTimeType() {
                 DSL.week(
                     functionProperties, DSL.literal(new ExprTimeValue("12:23:34")), DSL.literal(0)),
                 "week(TIME '12:23:34', 0)",
-                LocalDate.now(functionProperties.getQueryStartClock()).get(ALIGNED_WEEK_OF_YEAR)),
+                LocalDate.now(functionProperties.getQueryStartClock()).get(ALIGNED_WEEK_OF_YEAR)
```
Comment (penghuo) [inline core/src/test/java/org/opensearch/sql/expression/datetime/DateTimeFunctionTest.java:1239]: ignore this test case and add issue link? my concern is this temporary fix could fail again.

Comment (vamsi-amazon) [reply core/src/test/java/org/opensearch/sql/expression/datetime/DateTimeFunctionTest.java:1239]: Tied ignoring the coverage is falling below 100% percent.
It won't fail again this year. 2024 Jan 1 is Monday as a result week of the year is off by 1.

**Review by penghuo — APPROVED** [review]

**Review by derek-ho — APPROVED** [review]

**Review by YANG-DB — APPROVED** [review]

File: spark/src/main/antlr/FlintSparkSqlExtensions.g4 (hunk 1 above)
Comment (vamsi-amazon) [reply spark/src/main/antlr/FlintSparkSqlExtensions.g4:29]: These g4 files come from opensearch-spark. We always keep them up to date.

**Pull request merged**